### DISCUSSION TITLE
[Feature]: Duck option for audio from other apps on Android

### DISCUSSION THREAD

**Discussion by kehuang805** [TheWidlarzGroup/react-native-video#4983] [Ideas]
### What problem are you trying to solve?

Android currently only offers two options with `disableFocus` for either overriding or mixing outside audio, and there does not appear to be a way to duck audio from other apps with react-native-video.

iOS has the option to duck with the `mixWithOthers` property, but it is not present for Android, so it is not possible to have consistent behavior between operating systems currently.

### Proposed solution

Add a duck option for audio from other apps on Android, providing equivalent behavior to the iOS `mixWithOthers` property so that audio focus behavior is consistent across operating systems.

### Platforms

Android, iOS

[a comment by an automated account is not shown]

**Comment by paulrinaldi** [comment]
Apparently android has ducking built in somehow, we'd need to look at the code to see what we might need to do.
https://developer.android.com/media/optimize/audio-focus#automatic_ducking

**Comment by freeboub** [comment]
@kehuang805 I am currently working around audio focus. Can you please describe exactly the behavior you want to have ?

There are 3 props to handle audio: 
**ignoreSilentSwitch** is purely an ios feature
**mixWithOthers** and **disableFocus** nearly have the same behavior (in fact they should be merged)

"inherit" (default) --> pause playback when when other app use audio (same as disableFocus=false) 
"mix" - Audio from this video mixes with audio from other apps. (same as disableFocus=true)
"duck" - Reduces the volume of other apps while audio from this video plays. (Not available, but can be implemented)

**Comment by kehuang805** [comment]
@freeboub We would like `disableFocus` to have the "duck" option which is currently not available like you mentioned. I guess it does not necessarily have to be coupled to the `disableFocus` prop, but that is where the control for this behavior currently lives for android.

**Comment by valeriiamykhalova** [comment]
@kehuang805 @kehuang805 Hey! Have you been able to move forward with this issue? Would really appreciate `mixWithOthers` working on Android. If it's not the current priority, could you please share what the main complications are and where to look at?

**Comment by TWG-AI** [comment]
In v7 audio ducking is available via `player.mixAudioMode = 'duckOthers'`, which lowers other apps' audio while your video plays. The full set of modes is `'mixWithOthers' | 'doNotMix' | 'duckOthers' | 'auto'`.

> I'm an AI — verify platform behavior in the v7 docs.
